squads/dev-app/LuciaAuth/credentials.md

```diff
@@ -0,0 +1,458 @@
+		return Response.json("Email ou senha não válidos", { status: 400 });
+	}
+
+	const hashed_password = user.hashed_password;
+
+	if (!hashed_password) {
+		return Response.json("Unauthorized", { status: 401 });
+	}
+	const validPassword = bcrypt.compareSync(password, hashed_password);
+
+	if (!validPassword) {
+		return Response.json("Email ou senha não válidos", { status: 400 });
+	}
+
+	const session = await lucia.createSession(user.id, {});
+	const sessionCookie = lucia.createSessionCookie(session.id);
+	return Response.json(null, {
+		status: 302,
+		headers: {
+			Location: "/profile",
+			"Set-Cookie": sessionCookie.serialize(),
+		},
+	});
+}
+
+```
+===
+
+## Validar sessão/ Verificar usuário
+
+!!!
+Para fazer validações, o backend precisa verificar se a request de validação ocorreu por meio de um [CSRF.](https://www.treinaweb.com.br/blog/cross-site-request-forgery-csrf-e-abordagens-para-mitiga-lo)
+
+==- Exemplo Next
+Para o next esse ataque pode ser evitado utilizando o middleware.ts
+
+```ts middleware.ts
+import { verifyRequestOrigin } from "lucia";
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+export async function middleware(request: NextRequest): Promise<NextResponse> {
+	if (request.method === "GET") {
+		return NextResponse.next();
+	}
+	const originHeader = request.headers.get("Origin");
+	const hostHeader = request.headers.get("Host");
+	if (
+		!originHeader ||
+		!hostHeader ||
+		!verifyRequestOrigin(originHeader, [hostHeader])
+	) {
+		return new NextResponse(null, {
+			status: 403,
+		});
+	}
+	return NextResponse.next();
+}
+
+```
+===
+!!!
+
+Como diversas páginas podem necessitar do usuário, uma boa ideia é criar uma função para não precisar escrever tudo em várias páginas. A função abaixo retornará um `user` se ele estiver "logado".
+
+==- Função para pedir usuário
+```ts utils/getUser.ts
+import { cookies } from "next/headers";
+import { cache } from "react";
+
+import type { Session, User } from "lucia";
+import { lucia } from "../../auth/lucia";
+
+export const validateRequest = cache(
+	async (): Promise<
+		{ user: User; session: Session } | { user: null; session: null }
+	> => {
+		const sessionId = cookies().get(lucia.sessionCookieName)?.value ?? null;
+		if (!sessionId) {
+			return {
+				user: null,
+				session: null,
+			};
+		}
+
+		const result = await lucia.validateSession(sessionId);
+		try {
+			if (result.session && result.session.fresh) {
+				const sessionCookie = lucia.createSessionCookie(
+					result.session.id
+				);
+				cookies().set(
+					sessionCookie.name,
+					sessionCookie.value,
+					sessionCookie.attributes
+				);
+			}
+			if (!result.session) {
+				const sessionCookie = lucia.createBlankSessionCookie();
+				cookies().set(
+					sessionCookie.name,
+					sessionCookie.value,
+					sessionCookie.attributes
+				);
+			}
+		} catch {}
+		return result;
+	}
+);
+
+```
```

Comment:
getUSer -> faz uma descrição do que está sendo feito e explique (pega cookies -> verifica
se há dados -> verifica se dados da sessão dos cookies é valida -> se for, revalida sessão, seta cookies de novo e manda dados da sessão), explica tbm o que é o `session.fresh` rapidinho.